parser/charset/encoding.go

```diff
@@ -140,8 +141,40 @@ func (e *Encoding) EncodeInternal(dest, src []byte) []byte {
 	return dest
 }
 
+// validUTF8 checks whether there are illegal utf8 characters in []byte.
+func (e *Encoding) validUTF8(src []byte) ([]byte, error) {
```

Comment:
there is `utf8.Valid()`, we can avoid write it by ourself

Reply:
Implement this func for compatible err msg.